src/istio/values/values.yaml

```diff
@@ -6,3 +6,11 @@ meshConfig:
     holdApplicationUntilProxyStarts: true
     gatewayTopology:
       forwardClientCertDetails: SANITIZE
+  extensionProviders:
+    - name: "authservice"
+      envoyExtAuthzGrpc:
+        service: "authservice.authservice.svc.cluster.local"
+        port: "10003"
+
+pilot:
+  jwksResolverExtraRootCA: "" # todo: this needs to have the non-base64 encoded CA cert for keycloak
```

Comment:
To make a CA cert fully work with authservice we will need to set this or provide a different way for istiod to hit the jwks endpoint. This is necessary for jwtRules to work in the RequestAuthentication.

Reply:
Tested locally and `PILOT_JWT_ENABLE_REMOTE_JWKS` = `hybrid` env on istiod does work and is a pretty simple way to do. The tradeoff is that the sidecar on the workload (i.e. mission app) is the one hitting the jwks endpoint, so it needs egress to the keycloak service. I don't think there's any risk here?, and this is something we could easily generate as part of the authservice netpol additions. We discussed this yesterday but for reference [this is the issue](https://www.github.com/istio/istio/issues/40553) where that is mentioned as an approach. Would love @bburky's review on whether that makes sense and is safe to do...

The alternative of setting the CA for istiod + hitting keycloak on the external side is a bit of a pain since the istiod chart expects the CA as a "plain PEM" (not base64 encoded). It's generally easier, and more aligned with how we handle the gateway certs, if we expect it to be passed in base64 encoded. To make that work we could either load the certs via volume/mount on istiod ourselves (creating a secret outside of the upstream chart) or just contribute upstream to istio for them to support the CA in a secret/base64 encoded format instead of the current configmap. Not opposed to these options, but they don't feel as seamless.

Another random note - this is already identified as a need for us to manage providing CAs to other packages - https://github.com/defenseunicorns/uds-core/issues/464 - so in the future this might be something for pepr to manage and inject in different spots. Current state that wouldn't help us with istiod since pepr deploys after istio (we could still make it work, but would have to modify + cycle istiod after deploy rather than mutating on deploy).

Reply:
I think in-mesh egress from app sidecar to keycloak is fine. The AuthorizationPolicy DENY rules will still block `/admin*`, `/realms/master*`, `/metrics*` and `/realms/{{ .Values.realm }}/clients-registrations/*`, and I think that's all we care about. This doesn't really grant any additional access then.

Reply:
Resolving this thread - validated custom CA + certs on the latest commits.